shell/src/main/java/alluxio/cli/ValidateEnv.java

```diff
@@ -0,0 +1,392 @@
+      if (name != null && !taskName.startsWith(name)) {
+        continue;
+      }
+
+      System.out.format("Validating %s...", taskName);
+      if (task.validate()) {
+        System.out.println("OK");
+      } else {
+        System.out.println("Failed");
+        failureCount++;
+      }
+
+      validationCount++;
+    }
+
+    if (failureCount > 0) {
+      System.out.format("Validation failed. Total failures: %d.%n", failureCount);
+      return false;
+    }
+
+    if (validationCount == 0) {
+      System.out.format("No validation task matched name \"%s\".%n", name);
+      return false;
+    }
+
+    System.out.println("Validation succeeded.");
+    return false;
+  }
+
+  private static boolean validateWorkers(String name) throws InterruptedException {
+    return validateRemote(readNodeList("workers"), "worker", name);
+  }
+
+  private static boolean validateMasters(String name) throws InterruptedException {
+    return validateRemote(readNodeList("masters"), "master", name);
+  }
+
+  /**
+   * Prints the help message.
+   *
+   * @param message message before standard usage information
+   */
+  public static void printHelp(String message) {
+    System.err.println(message);
+    HelpFormatter help = new HelpFormatter();
+    help.printHelp(USAGE, OPTIONS);
+  }
+
+  /**
+   * Validate environment.
+   *
+   * @param args list of arguments
+   * @return 0 on success, 1 on failures
+   */
+  public static int validate(String... args) throws InterruptedException {
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd;
+
+    try {
+      cmd = parser.parse(OPTIONS, args, true /* stopAtNonOption */);
+    } catch (ParseException e) {
+      printHelp("Unable to parse input args: " + e.getMessage());
+      return 1;
+    }
+
+    Preconditions.checkNotNull(cmd, "Unable to parse input args.");
+    args = cmd.getArgs();
+    if (args.length < 1) {
+      printHelp("Target not specified.");
+      return 1;
+    }
+
+    if (args.length > 2) {
+      printHelp("More arguments than expected.");
+      return 1;
+    }
+
+    String target = args[0];
+    String name = args.length > 1 ? args[1] : null;
+
+    boolean success = false;
+    switch (target) {
+      case "local":
+      case "worker":
+      case "master":
+        success = validateLocal(target, name);
+        break;
+      case "all":
+        success = validateMasters(name) || validateWorkers(name);
+        break;
+      case "workers":
+        success = validateWorkers(name);
+        break;
+      case "masters":
+        success = validateMasters(name);
+        break;
+      default:
+        printHelp("Invalid target.");
+    }
+
+    return success ? 0 : 1;
+  }
+
+  /**
+   * Validate Alluxio environment.
+   *
+   * @param args the arguments to specify which validation tasks to run
+   */
+  public static void main(String[] args) throws InterruptedException {
+    System.exit(validate(args));
```

Comment:
I suggest you distinguish invalid environment return value (-1) from incorrect invocation return value (-2) and when -2 is return, you print a usage message as well ... -1 and -2 should be defined as named constants